streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java

```diff
@@ -57,31 +60,61 @@
     private final Consumed<Integer, String> consumed = Consumed.with(intSerde, stringSerde);
 
     @Before
-    public void setUp() throws IOException {
+    public void setUp() {
         stateDir = TestUtils.tempDirectory("kafka-test");
     }
 
+    @Test
+    public void shouldLogAndMeterOnSkippedRecords() {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final KStream<String, Integer> left = builder.stream("left", Consumed.with(stringSerde, intSerde));
+        final KStream<String, Integer> right = builder.stream("right", Consumed.with(stringSerde, intSerde));
+
+        left.join(
+            right,
+            new ValueJoiner<Integer, Integer, Integer>() {
+                @Override
+                public Integer apply(final Integer value1, final Integer value2) {
+                    return value1 + value2;
+                }
+            },
+            JoinWindows.of(100),
+            Joined.with(stringSerde, intSerde, intSerde)
+        );
+
+        final LogCaptureAppender appender = LogCaptureAppender.createAndRegister();
+        driver.setUp(builder, stateDir);
+        driver.process("left", "A", null);
+        LogCaptureAppender.unregister(appender);
+
+        assertThat(appender.getMessages(), hasItem("Skipping record due to null key or value. key=[A] value=[null] topic=[left] partition=[-1] offset=[-1]"));
+
+        assertEquals(1.0, MetricTestUtil.getMetricByName(driver.context().metrics().metrics(), "skipped-records-total").metricValue());
+        System.out.println("");
```

Comment:
is this line intentional?

Reply:
oops. I put it there when I needed a place for a breakpoint. Sorry!